src/withApiData.test.ts

```diff
@@ -1,57 +1,156 @@
+        expect(testOne).toBe(true);
+    });
+    test('The request status or params has been changed, or bindings are set correctly and status is not changed (should be false)', () => {
+        const testTwo = shouldPerformApiRequest(
+            getProps(endpoint, true, {}, 'success'),
+            getProps(endpoint, true, {}, 'success'), bindings, 'getData');
+        expect(testTwo).toBe(false);
+    });
+    test('The request status or params has been changed, when new params are changed (should perform)', () => {
+        const testTree = shouldPerformApiRequest(
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'success'),
+            getProps(endpoint, true, { getData: { two: 'two' } }, 'success'), bindings, 'getData');
+        expect(testTree).toBe(true);
+    });
+    test('The request status or params has been changed, when params are set and status has changed', () => {
+        const testFour = shouldPerformApiRequest(
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'ready'),
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'success'), bindings, 'getData');
+        expect(testFour).toBe(true);
+    });
+    test('The request status or params has been changed, when params are set but status hasn\'t changed', () => {
+        const testFive = shouldPerformApiRequest(
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'success'),
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'success'), bindings, 'getData');
+        expect(testFive).toBe(false);
+    });
+    test('The request status or params has been changed, when status is undefined', () => {
+        const testSix = shouldPerformApiRequest(
+            getProps(endpoint, true, { getData: { one: 'one' } }, 'ready'),
+            getProps(endpoint, true, {}), bindings, 'getData');
+        expect(testSix).toBe(true);
+    });
+    test('The request status or params has been changed, when config is undefined an old request is undefined', () => {
+        const testSeven = shouldPerformApiRequest(
+            getProps(endpoint, true, { getData: { one: 'one' } }),
+            getProps(endpoint, false, {}, 'ready'), bindings, 'getData');
+        expect(testSeven).toBe(true);
+    });
+    test('The request status or params has been changed, when config is undefined and new config is created', () => {
+        const testEight = shouldPerformApiRequest(
+            getProps(endpoint, false, {}),
+            getProps(endpoint, true, { getData: {} }), bindings, 'getData');
+        expect(testEight).toBe(true);
+    });
+});
 
+describe('shouldAutoTrigger', () => {
+    test('GET triggered', () => {
+        const testAutoTrigger = shouldAutoTrigger(
+            getProps(endpoint, true, emptyParams, 'ready', { method: 'GET' }), 'getData');
+        expect(testAutoTrigger).toBe(true);
+    });
+    test('POST does not trigger', () => {
```

Comment:
could you merge post/put/delete/patch into 1 test, since it should behave the same, so one test that tests all should suffice. i.e. with a loop through ['POST, 'PUT', etc] (better readability, maintainability, lower chance of mistakes)